src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/assist/CompletionManagerBase.java

```diff
@@ -57,6 +57,11 @@
 public abstract class CompletionManagerBase
       implements CompletionRequestContext.Host
 {
+   public abstract void goToHelp();
```

Comment:
Given the empty implementation of these methods in various subclasses (with the NYI comments), would it be more appropriate to have empty implementations here in CompletionManagerBase instead of making these abstract?

Reply:
My main reason to keep these abstract is just that it's "easier" to implement a single class method, versus overriding an existing class method. E.g. as an author of a subclass, I might need to consider:

1. Which methods should I override to achieve some functionality? (need to potentially look at all overridable methods in base class, versus trusting the base class to "tell" you which methods are abstract and so require implementations)
2. Should I call the superclass method? If so, do I call it first, or last, or something else? (sometimes it's obvious, sometimes it's not; either way, you were forced to think about it)

Those considerations are moot for pure abstract methods, and so I think are then easier to reason about. There's a little bit more code, but IDEs auto-generate that for you so it's not as much of a concern.